packages/react-query/src/shared/proxy/utilsProxy.ts

```diff
@@ -323,6 +337,17 @@ export function createReactQueryUtilsProxy<
         getInfiniteData: () => context.getInfiniteQueryData(queryKey),
       };
 
+      if (overrides?.[utilName]) {
+        const override = overrides?.[utilName]?.({
+          queryKey: queryKey as unknown as TRPCQueryKey,
+          path,
+          input,
+        });
+        if (override instanceof Promise) {
+          return override.then(() => contextMap[utilName]());
+        }
+      }
```

Comment:
is this the best way to handle this lol

the callback in `createRecursiveProxy` can't be async since then sync methods like `getData` returns a promise

the override can be both sync / async and we wanna make sure that finishes running before the util runs.